File: javascript-version/customer.js

function newCustomer(name) {
  var rentals = [];
  
  function formatMoney(money) {
    return money.toFixed(1);
  };
  
  return {
    addRental: function(rental) {
      rentals.push(rental);
    },

    getName: function() {
      return name;
    },
    
    statement: function() {
      var totalAmount = 0;
      var frequentRenterPoints  = 0;
      var result = "Rental Record for " + this.getName() + "\n";

      for (var i=0; i<rentals.length; i++) {
        var each = rentals[i];
        var thisAmount = 0;

        // determines the amount for each line
        switch (each.getMovie().getPriceCode()) {
          case Movie.REGULAR:
            thisAmount += 2;
            if (each.getDaysRented () > 2)
              thisAmount += (each.getDaysRented () - 2) * 1.5;
            break;
          case Movie.NEW_RELEASE:
            thisAmount += each.getDaysRented () * 3;
            break;
          case Movie.CHILDRENS:
            thisAmount += 1.5;
            if (each.getDaysRented () > 3)
              thisAmount += (each.getDaysRented () - 3) * 1.5;
            break;
        }

        frequentRenterPoints++;

        if (each.getMovie().getPriceCode() == Movie.NEW_RELEASE 
            && each.getDaysRented() > 1)
          frequentRenterPoints++;

        result += "\t" + each.getMovie().getTitle() + "\t"
                  + formatMoney(thisAmount) + "\n";
        totalAmount += thisAmount;      
      }

      result += "You owed " + formatMoney(totalAmount) + "\n";
      result += "You earned " + frequentRenterPoints + " frequent renter points\n";

      return result;
    },
  };
}

File: csharp-version/VideoStore/Customer.cs
using System;
using System.Collections.Generic;

namespace VideoStore
{
	public class Customer 
	{		
		private string name;
		private List<Rental> rentals = new List<Rental> ();

		public Customer (string name) {
			this.name = name;
		}

		public void AddRental(Rental rental) {
			rentals.Add(rental);
		}

		public string GetName () {
			return name;
		}

		public string Statement () {
			double 				totalAmount 			= 0;
			int					frequentRenterPoints 	= 0;
			var 				rentals 				= this.rentals.GetEnumerator();
			string 				result 					= "Rental Record for " + GetName () + "\n";

			while (rentals.MoveNext()) {
				double 		thisAmount = 0;
				Rental 		each = rentals.Current;

				// determines the amount for each line
				switch (each.GetMovie ().GetPriceCode ()) {
					case Movie.REGULAR:
					thisAmount += 2;
					if (each.GetDaysRented () > 2)
						thisAmount += (each.GetDaysRented () - 2) * 1.5;
					break;
					case Movie.NEW_RELEASE:
					thisAmount += each.GetDaysRented () * 3;
					break;
					case Movie.CHILDRENS:
					thisAmount += 1.5;
					if (each.GetDaysRented () > 3)
						thisAmount += (each.GetDaysRented () - 3) * 1.5;
					break;
				}

				frequentRenterPoints++;

				if (each.GetMovie ().GetPriceCode () == Movie.NEW_RELEASE 
				    && each.GetDaysRented () > 1)
					frequentRenterPoints++;

				result += "\t" + each.GetMovie ().GetTitle () + "\t"
					+ thisAmount.ToString("0.0") + "\n";
				totalAmount += thisAmount;

			}

			result += "You owed " + totalAmount.ToString("0.0") + "\n";
			result += "You earned " + frequentRenterPoints + " frequent renter points\n";


			return result;
		}
	}
}

